odc/stac/_mdtools.py

```diff
@@ -333,36 +333,44 @@ def alias_map_from_eo(item: pystac.item.Item, quiet: bool = False) -> Dict[str,
     """
     Generate mapping ``common name -> canonical name``.
 
-    For all unique common names defined on the Item eo extension record mapping to the canonical
-    name. Non-unique common names are ignored with a warning unless ``quiet`` flag is set.
+    For all unique common names defined on the Item's assets via the eo
+    extension, record a mapping to the asset key ("canonical name"). Non-unique
+    common names are ignored with a warning unless ``quiet`` flag is set.
 
     :param item: STAC :class:`~pystac.item.Item` to process
     :param quiet: Do not print warning if duplicate common names are found, defaults to False
     :return: common name to canonical name mapping
     """
-    try:
-        bands = EOExtension.ext(item, add_if_missing=False).bands
-    except pystac.errors.ExtensionNotImplemented:
-        return {}
-
-    if bands is None:
-        return {}  # pragma: no cover
-
-    common_names: Dict[str, Set[str]] = {}
-    for band in bands:
-        common_name = band.common_name
-        if common_name is not None:
-            common_names.setdefault(common_name, set()).add(band.name)
-
-    def _aliases(common_names):
-        for alias, bands in common_names.items():
+    aliases: Dict[str, Set[str]] = {}
+    for key, asset in item.assets.items():
+        try:
+            eo = EOExtension.ext(asset)
+        except pystac.errors.ExtensionNotImplemented:
```

Comment:
Just to confirm, this exception will only happen is entire item has not EO so early exit is fine

Reply:
Correct, since there is no such thing as asset-level extensions.